libblkid/src/probe.c

```diff
@@ -1159,8 +1185,10 @@ int blkid_probe_set_device(blkid_probe pr, int fd,
 	}
 # endif
 
-	DBG(LOWPROBE, ul_debug("ready for low-probing, offset=%"PRIu64", size=%"PRIu64", zonesize=%"PRIu64,
-				pr->off, pr->size, pr->zone_size));
+	pr->io_size = blkid_get_io_size(fd);
```

Comment:
It would be better to call blkid_get_io_size() only `if (S_ISBLK(sb.st_mode))`, maybe also add `!is_floppy` and `!blkid_probe_is_tiny(pr)`. We want to keep blkid_probe_set_device() as fast as possible.

Suppose you have 10K files, devices, etc. to probe. ;-)

Reply:
Naively I would guess it isn't that bad.
But all the other checks are the same, so let's do it here, too.